mlflow/server/handlers.py

```diff
@@ -164,6 +165,22 @@ def _get_experiment():
     return response
 
 
+@catch_mlflow_exception
+def _get_experiment_by_name():
+    request_message = _get_request_message(GetExperimentByName())
+    response_message = GetExperimentByName.Response()
+    store_exp = _get_store().get_experiment_by_name(request_message.experiment_name)
```

Comment:
Kind of funny - in the OSS server, we call the AbstractStore implementation, which ([at least based on existing precedent](https://github.com/mlflow/mlflow/blob/b587dc4c8913ba5069871b984d31a06277e4c80a/mlflow/store/abstract_store.py#L65)) is supposed to return `None` when no matching experiment is found. This is currently the case in the FileStore ([see relevant test](https://github.com/mlflow/mlflow/blob/b587dc4c8913ba5069871b984d31a06277e4c80a/tests/store/test_file_store.py#L160)) & SqlAlchemyStore (this PR adds a relevant test in `test_sqlalchemy_store.py`), and we have an e2e test that the REST client handles the OSS server's RESOURCE_DOES_NOT_EXIST response [here](https://github.com/mlflow/mlflow/pull/1775/files#diff-963fd074b87c2dfb21917f87b1a82628R467)